src/goto-symex/goto_state.h

```diff
@@ -28,8 +28,15 @@ class goto_statet
   /// Distance from entry
   unsigned depth = 0;
 
+protected:
   symex_level2t level2;
```

Comment:
What about making current_names private in symex levels instead? (or both)

Reply:
Both sounds reasonable to me, but suggest doing further refactoring elsewhere, this PR has already dragged on a fair while